11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+		IGPUCommandBuffer::SPipelineBarrierDependencyInfo bufferPipelineBarrierInfo = {};
+		decltype(bufferPipelineBarrierInfo)::buffer_barrier_t bufBarrier = {};
+		bufferPipelineBarrierInfo.bufBarriers = { &bufBarrier, 1u };
+
+		// First axis FFT writes to colMajorBuffer
+		bufBarrier.range.buffer = m_colMajorBuffer;
+
+		// Wait for first compute write (first axis FFT) before next compute read (second axis FFT)
+		bufBarrier.barrier.dep.srcStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT;
+		bufBarrier.barrier.dep.srcAccessMask = ACCESS_FLAGS::MEMORY_WRITE_BITS;
+		bufBarrier.barrier.dep.dstStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT;
+		bufBarrier.barrier.dep.dstAccessMask = ACCESS_FLAGS::MEMORY_READ_BITS;
+
+		m_computeCmdBuf->pipelineBarrier(asset::E_DEPENDENCY_FLAGS(0), bufferPipelineBarrierInfo);
+		// Now comes Second axis FFT + Conv + IFFT
+		m_computeCmdBuf->bindComputePipeline(m_lastAxisFFT_convolution_lastAxisIFFTPipeline.get());
+		m_computeCmdBuf->bindDescriptorSets(asset::EPBP_COMPUTE, m_lastAxisFFT_convolution_lastAxisIFFTPipeline->getLayout(), 0, 1, &m_lastAxisFFT_convolution_lastAxisIFFTDescriptorSet.get());
+		// We need to pass the log of number of workgroups as a push constant now
+		uint32_t numWorkgroups = core::roundUpToPoT(marginSrcDim.height) / 2;
+		uint32_t workgroupsLog2 = std::bit_width(numWorkgroups) - 1;
+		m_computeCmdBuf->pushConstants(m_lastAxisFFT_convolution_lastAxisIFFTPipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, offsetof(PushConstantData, numWorkgroupsLog2), sizeof(pushConstants.numWorkgroupsLog2), &workgroupsLog2);
+
+		m_computeCmdBuf->dispatch(numWorkgroups, 1, 1);
+
+		// Recycle pipeline barrier, only have to change which buffer we need to wait to be written to
+		bufBarrier.range.buffer = m_rowMajorBuffer;
+		m_computeCmdBuf->pipelineBarrier(asset::E_DEPENDENCY_FLAGS(0), bufferPipelineBarrierInfo);
+
+		// Finally run the IFFT on the first axis
+		m_computeCmdBuf->bindComputePipeline(m_firstAxisIFFTPipeline.get());
+		m_computeCmdBuf->bindDescriptorSets(asset::EPBP_COMPUTE, m_firstAxisIFFTPipeline->getLayout(), 0, 1, &m_firstAxisIFFTDescriptorSet.get());
+		// One workgroup per 2 columns
+		m_computeCmdBuf->dispatch(srcDim.width / 2, 1, 1);
+		m_computeCmdBuf->end();
+
+		// Submit to queue and add sync point
+		{
+			const IQueue::SSubmitInfo::SCommandBufferInfo cmdbufInfo =
+			{
+				.cmdbuf = m_computeCmdBuf.get()
+			};
+			const IQueue::SSubmitInfo::SSemaphoreInfo signalInfo =
+			{
+				.semaphore = m_timeline.get(),
+				.value = ++semaphorValue,
+				.stageMask = asset::PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT
+			};
+
+			const IQueue::SSubmitInfo submitInfo = {
+				.waitSemaphores = {},
+				.commandBuffers = {&cmdbufInfo,1},
+				.signalSemaphores = {&signalInfo,1}
+			};
+
+			m_api->startCapture();
+			m_queue->submit({ &submitInfo,1 });
+			m_api->endCapture();
+		}
+
+		// Kill after one iteration for now
+		m_keepRunning = false;
```

Comment:
do the whole swapchain/window thing, look at Nipun's example or ex 27

Reply:
P.S. you either need a blit or storage image into the swapchain image, either way you need a pipeline after dispatch to transition the output image from GENERAL to PRESENT_SOURCE or do some other stuff in case of a blit